src/Waargonaut/Decode.hs

```diff
@@ -91,91 +91,100 @@ module Waargonaut.Decode
-import           Control.Monad.Reader                      (ReaderT (..), ask,
-                                                            runReaderT)
-import           Control.Monad.State                       (MonadState)
-
-import           Control.Error.Util                        (note)
-import           Control.Monad.Error.Hoist                 ((<!?>), (<?>))
-
-import           Data.Either                               (Either (..))
-import qualified Data.Either                               as Either (either)
-import           Data.Foldable                             (Foldable, elem,
-                                                            foldl, foldr)
-import           Data.Function                             (const, flip, ($),
-                                                            (&))
-import           Data.Functor                              (fmap, (<$), (<$>))
-import           Data.Functor.Alt                          ((<!>))
-import           Data.Functor.Identity                     (Identity)
-import           Data.Monoid                               (mempty)
-import           Data.Scientific                           (Scientific)
-
-import           Data.List.NonEmpty                        (NonEmpty ((:|)))
-import           Data.Maybe                                (Maybe (..),
-                                                            fromMaybe, maybe)
-import           Natural                                   (Natural, replicate,
-                                                            successor', zero')
-
-import           Data.Text                                 (Text)
-
-import           Data.ByteString                           (ByteString)
-import qualified Data.ByteString.Char8                     as BS8
-import qualified Data.ByteString.Lazy                      as BL
-
-import           HaskellWorks.Data.Positioning             (Count)
-import qualified HaskellWorks.Data.Positioning             as Pos
-
-import qualified HaskellWorks.Data.BalancedParens.FindOpen as BP
-
-import           HaskellWorks.Data.Bits                    ((.?.))
-import           HaskellWorks.Data.TreeCursor              (TreeCursor (..))
-
-import           HaskellWorks.Data.Json.Cursor             (JsonCursor (..))
-import qualified HaskellWorks.Data.Json.Cursor             as JC
-
-import           Waargonaut.Decode.Error                   (AsDecodeError (..),
-                                                            DecodeError (..),
-                                                            Err (..))
-import           Waargonaut.Decode.ZipperMove              (ZipperMove (..))
-import           Waargonaut.Types                          (Json)
-
-import qualified Waargonaut.Decode.Internal                as DI
+import           GHC.Word                                       (Word64)
+
+import           Control.Lens                                   (Cons, Lens',
```

Comment:
I've given up trying to stylish haskell to 80 chars. I've bumped mine to 120 because this kind of formatting gets silly. Haha.